File: paper_routing/hnn_fpga/hnn_fpga.m
function [V] = hnn_fpga(U, l)
N1 = 8;
V = zeros(N1 * N1, 1);
for x = 1:N1
    for i = 1:N1
        if x == i
            V((x-1) * N1 + i) = 0;
        else
            V((x-1) * N1 + i) = 1 / (1 + exp(-1 * U((x-1) * N1 + i) * l));
        end
    end
end

File: paper_routing/hnn_fpga/hnn_fpga.cpp
#include <math.h>
#define N1 8    //Cantidad de nodos

void hnn_fpga(float U_1[N1 * N1], float l, float V[N1 * N1])
{
    for(int x = 0; x < N1; x++)
    {
        for(int i = 0; i < N1; i++)
        {
            if (x == i)
            {
                V[x * N1 + i] = 0;
            } else {
                V[x * N1 + i] = 1 / (1 + expf(-1 * U_1[x * N1 + i] * l));
            }
        }
    }
}